sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClient.java

```diff
@@ -165,6 +251,47 @@
             .switchIfEmpty(Mono.defer(() -> EMPTY_BYTE_STRING_MONO));
     }
 
+    /**
+     * Aggregate InputStream to single okio.ByteString.
+     *
+     * @param inputStream the InputStream to aggregate
+     * @return Aggregated ByteString
+     */
+    private static ByteString toByteString(InputStream inputStream) {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        int nRead;
+        byte[] buffer = new byte[8192];
+        try (InputStream closeableInputStream = inputStream) {
```

Comment:
We should look at creating a utility method in Core that can optimize InputStream to byte[] conversion based on the type of InputStream

Reply:
yeah, that be neat. There's `InputStream.readAllBytes()` with some interesting code since Java 9. So I guess it'd be something like we have for reflections perhaps.